pkg/issuer/acme/dns/azuredns/azuredns.go

```diff
@@ -96,6 +126,39 @@ func getAuthorization(env azure.Environment, clientID, clientSecret, subscriptio
 		opt.IdentityResourceID = managedIdentity.ResourceID
 	}
 
+	// Use Workload Identity if present
+	if os.Getenv("AZURE_FEDERATED_TOKEN_FILE") != "" {
+		spt, err := getFederatedSPT(env, opt)
+		if err != nil {
+			return nil, err
+		}
+
+		// adal does not offer methods to dynamically replace a federated token, thus we need to have a wrapper to make sure
+		// we're using up-to-date secret while requesting an access token
+		var refreshFunc adal.TokenRefresh = func(context context.Context, resource string) (*adal.Token, error) {
+			newSPT, err := getFederatedSPT(env, opt)
+			if err != nil {
+				return nil, err
+			}
+
+			// Need to call Refresh(), otherwise .Token() will be empty
+			err = newSPT.Refresh()
+			if err != nil {
+				return nil, err
+			}
```

Comment:
Calling `Refresh` sends the K8S ServiceAccountToken (JWT) to Azure's Oauth endpoint and gets back an Oauth AccessToken and RefreshToken, right?

Maybe add an additional comment explaining that this results in a roundtrip to the Azure auth API.

Reply:
When there's a special label in place (e.g. on a `ServiceAccount`: `azure.workload.identity/use: "true"`), a mutation webhook modifies a pod's spec to add a projected volume with a federated token. The path to the token is always the same, it's just a concatenation of these two constants:
```
	TokenFilePathName             = "azure-identity-token"
	TokenFileMountPath            = "/var/run/secrets/azure/tokens" // #nosec
```
https://github.com/Azure/azure-workload-identity/blob/231c6be82969efb51f0539397e6ab268c1eb8eca/pkg/webhook/consts.go#L54
To avoid hardcoding, the apps are supposed to find it through `AZURE_FEDERATED_TOKEN_FILE` env.
The token gets updated once an hour by default.

This token is sent to the Oauth endpoint. Response contains only an `AccessToken` (valid for 24h), `RefreshToken` is not present.

Reply:
Added more comments as requested.